[Bug]: test1 error

### Brief problem description

;,;;,

### Steps to Reproduce

;;.;.;.;

### Expected Result

idk

### Actual Result

FAILED tests/test_items.py::test_get_item_1 - assert 404 == 200

--- a/src/app/routers/items.py
+++ b/src/app/routers/items.py
@@ -36,7 +36,7 @@ def get_course(course_id: str):
     if course is not None:
         return course
 
-    raise HTTPException(status_code=404, detail="Course not found")
+    raise HTTPException(status_code=400, detail="Course not found")
 
 
 @router.get("/course/{course_id}/lab/{lab_id}", response_model=Lab)

--- a/src/app/services/item_service.py
+++ b/src/app/services/item_service.py
@@ -228,7 +228,7 @@ def get_item_by_id_dfs_iterative(
 
                     for task in lab.tasks:
                         counter += 1
-                        if lab.id == item_id:
+                        if task.id == item_id:
                             return FoundItem(task, counter)
 
                         for step in task.steps:

--- a/tests/test_items.py
+++ b/tests/test_items.py
@@ -50,6 +50,7 @@ def test_get_item_1(client: TestClient):
         task_id="lab-02-run-local",
         step_id="lab-02-run-local-venv",
         order="pre",
+        
     )
 
 

